src/simulation/m_riemann_solvers.fpp

```diff
@@ -560,6 +549,42 @@ contains
                                 end do
                             end if
 
+                            ! elastic energy update
```

Comment:
can you keep this in a separate branch until its ready to merge? i don't want to leave big patches of commented features in master.